src/components/PostDetails.tsx

```diff
@@ -1,117 +1,253 @@
+    name: false,
+    email: false,
+    body: false,
+  });
+
+  const [error, setError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isWriting, setIsWriting] = useState<boolean>(false);
+
+  const clearFormField = useCallback(() => {
+    setNewComment({
+      name: '',
+      email: '',
+      body: '',
+    });
+
+    setFieldError({
+      name: false,
+      email: false,
+      body: false,
+    });
+  }, [comments]);
+
+  const newCommentId = useCallback(() => {
+    const commentIds = comments.map(comment => comment.id);
+
+    return (Math.max(...commentIds) + 1);
+  }, [comments]);
+
+  const isEmailValid = useCallback((email: string) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+    return emailRegex.test(email.trim());
+  }, [newComment]);
+
+  const validateFields = useCallback((
+    innerNewComment: CommentData, innerFieldErrors: FieldErrors,
+  ) => {
+    const { name, email, body } = innerNewComment;
+    const newFieldErrors = { ...innerFieldErrors };
+
+    if (!name.trim()) {
+      newFieldErrors.name = true;
+    }
+
+    if (!isEmailValid(email)) {
+      newFieldErrors.email = true;
+    }
+
+    if (!body.trim()) {
+      newFieldErrors.body = true;
+    }
+
+    return setFieldError(newFieldErrors);
+  }, [newComment]);
+
+  const addComment = useCallback(async (
+    event: FormEvent<HTMLButtonElement>,
+  ) => {
+    event.preventDefault();
+
+    const { name, email, body } = newComment;
+
+    if (!name.trim() || !isEmailValid(email) || !body.trim()) {
+      validateFields(newComment, fieldErrors);
+    } else if (selectedPost?.id) {
+      const newCommentObject: Comment = {
+        id: newCommentId(),
+        postId: selectedPost.id,
+        name: newComment.name.trim(),
+        email: newComment.email.trim(),
+        body: newComment.body.trim(),
+      };
+
+      setComments([...comments, newCommentObject]);
+
+      try {
+        await createComment(newCommentObject);
+      } catch (innerError) {
+        setError(true);
+      } finally {
+        setNewComment({ ...newComment, body: '' });
+      }
+    }
+  }, [newComment, fieldErrors, selectedPost, comments]);
+
+  const imputHandler = useCallback((
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+    innerNewComment: CommentData,
+  ) => {
+    event.preventDefault();
+    const { name, value } = event.target;
+
+    if (value.length === 0) {
+      setFieldError({ ...fieldErrors, [name]: true });
+    } else {
+      setFieldError({ ...fieldErrors, [name]: false });
+    }
+
+    return setNewComment({ ...innerNewComment, [name]: value });
+  }, [newComment]);
+
+  const deleteCommentHandler = useCallback(async (commentId: number, event) => {
+    event.preventDefault();
+    setComments(comments.filter(comment => comment.id !== commentId));
+
+    try {
+      await deleteComment(commentId);
+    } catch (innerError) {
+      throw new Error();
```

Comment:
Instead of throwing a new Error, consider setting the 'error' state to true: setError(true).